subsys/bluetooth/mesh/shell/shell_scheduler_cli.c

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <stdlib.h>
+#include <bluetooth/mesh/models.h>
+#include <shell/shell.h>
+#include <sys/math_extras.h>
+
+#include "mesh/net.h"
+#include "mesh/access.h"
+#include "../model_utils.h"
+
+#include "shell_utils.h"
+
+static struct bt_mesh_model *mod;
+static struct bt_mesh_schedule_entry set_entry;
+
+static int cmd_get(const struct shell *shell, size_t argc, char *argv[])
+{
+	if (!mod && !shell_model_first_get(BT_MESH_MODEL_ID_SCHEDULER_CLI, &mod)) {
+		return -ENODEV;
+	}
+
+	struct bt_mesh_scheduler_cli *cli = mod->user_data;
+	uint16_t rsp;
+
+	int err = bt_mesh_scheduler_cli_get(cli, NULL, &rsp);
+
+	if (!err) {
+		shell_print(shell, "Status: %d", rsp);
+	}
+
+	return err;
+}
+
+static void schedule_print(const struct shell *shell, int err, struct bt_mesh_schedule_entry rsp)
+{
+	if (!err) {
+		shell_print(shell,
+			    "Date: %02d.%02d.%02d, Time: %d:%d:%d,\n"
+			    "Weekday: %d, action: %d, trans time: %d, scene num: %d",
+			    rsp.day, (u32_count_trailing_zeros(rsp.month) + 1), rsp.year, rsp.hour,
+			    rsp.minute, rsp.second, (u32_count_trailing_zeros(rsp.day_of_week) + 1),
+			    rsp.action, model_transition_decode(rsp.transition_time),
+			    rsp.scene_number);
+	}
+}
+
+static int cmd_action_get(const struct shell *shell, size_t argc, char *argv[])
+{
+	uint8_t idx = (uint8_t)strtol(argv[1], NULL, 0);
+
+	if (!mod && !shell_model_first_get(BT_MESH_MODEL_ID_SCHEDULER_CLI, &mod)) {
+		return -ENODEV;
+	}
+
+	struct bt_mesh_scheduler_cli *cli = mod->user_data;
+	struct bt_mesh_schedule_entry rsp;
+
+	int err = bt_mesh_scheduler_cli_action_get(cli, NULL, idx, &rsp);
+
+	schedule_print(shell, err, rsp);
+	return err;
+}
+
+static int cmd_action_ctx_set(const struct shell *shell, size_t argc, char *argv[])
```

Comment:
I guess the scene number might be excluded if action above On Off model.

Reply:
Im not quite sure if I got your point, but it seems like the scene param is packed and sent regardless for the action set calls. Could you elaborate?

Reply:
I mean if actions 0x00\0x01 (Turn Off\On) it is worth to omit scene number instead of providing fake 0 (logic `No scene`).
but up to you, if you feel it is worth to provide the scene number in any case just skip this comment.